src/include/btree_cmp_inline.h

```diff
@@ -166,6 +292,132 @@ __wt_compare_bounds(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_ITEM *key, u
     return (0);
 }
 
+#ifdef HAVE_X86INTRIN_H
+/*
+ * __lex_compare_skip_ge_16 --
+ *     Lexicographic comparison routine for data greater than or equal to 16 bytes, skipping leading
+ *     bytes. Returns: < 0 if user_item is lexicographically < tree_item = 0 if user_item is
+ *     lexicographically = tree_item > 0 if user_item is lexicographically > tree_item We use the
+ *     names "user" and "tree" so it's clear in the btree code which the application is looking at
+ *     when we call its comparison function. Some platforms like ARM offer dedicated instructions
+ *     for reading 16 bytes at a time, allowing for faster comparisons. Some platforms like ARM
+ *     offer dedicated instructions for reading 16 bytes at a time, allowing for faster comparisons.
+ */
+static WT_INLINE int
+__lex_compare_skip_ge_16(
+  const uint8_t *ustartp, const uint8_t *tstartp, size_t len, int lencmp, size_t *matchp)
+{
+    __m128i res_eq, t, u;
+    uint64_t t64, tfirst, u64, ufirst;
+    size_t match;
+    const uint8_t *tendp, *treep, *uendp, *userp;
+    bool firsteq;
+
+    match = *matchp;
+    uendp = ustartp + len;
+    tendp = tstartp + len;
+
+    /* skip 16 matching bytes at a time, starting at first possible difference. */
+    for (userp = ustartp + match, treep = tstartp + match; uendp - userp > WT_VECTOR_SIZE;
+         userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
+        u = _mm_loadu_si128((const __m128i *)userp);
+        t = _mm_loadu_si128((const __m128i *)treep);
+        res_eq = _mm_cmpeq_epi8(u, t);
+        if (_mm_movemask_epi8(res_eq) != 65535) {
+            match = (size_t)(userp - ustartp);
+            goto final128;
+        }
+    }
+
+    /*
+     * Rewind until there is exactly 16 bytes left. We know we started with at least 16, so we are
+     * still in bound.
+     */
+    u = _mm_loadu_si128((const __m128i *)(uendp - WT_VECTOR_SIZE));
+    t = _mm_loadu_si128((const __m128i *)(tendp - WT_VECTOR_SIZE));
+    match = (size_t)(uendp - ustartp) - WT_VECTOR_SIZE;
+
+final128:
+    ufirst = (uint64_t)_mm_extract_epi64(u, 0);
+    tfirst = (uint64_t)_mm_extract_epi64(t, 0);
+    firsteq = ufirst == tfirst;
+    u64 = firsteq ? (uint64_t)_mm_extract_epi64(u, 1) : ufirst;
+    t64 = firsteq ? (uint64_t)_mm_extract_epi64(t, 1) : tfirst;
+    match += firsteq ? sizeof(uint64_t) : 0;
+
+    /* x86 is little endian so we need to flip the bytes. */
+    u64 = __wt_bswap64(u64);
+    t64 = __wt_bswap64(t64);
+
+    match += (size_t)__builtin_clzll(u64 ^ t64) / 8;
```

Comment:
```suggestion
    WT_LEADING_ZEROS((u64 ^ t64), leading_zero_bytes);
    match += leading_zero_bytes;
```
also in the non-`HAVE_X86INTRIN_H` version of this function.

Reply:
This seems less efficient?

Reply:
done